components/script/dom/htmlmediaelement.rs

```diff
@@ -602,6 +690,7 @@ impl HTMLMediaElement {
                     .. RequestInit::default()
                 };
 
+                self.setup_media_player();
```

Comment:
I suspect that this is not the best place to setup the media player. I need to properly read the spec, but I would say that we should be setting it up a bit earlier, and ideally handling any potential setup failure.